clients/ops/admin-ui/src/features/subject-request/events-and-logs/EventDetails.tsx

```diff
@@ -0,0 +1,164 @@
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { useDisclosure } from "@chakra-ui/react";
+import {
+  Box,
+  Divider,
+  Drawer,
+  DrawerBody,
+  DrawerContent,
+  DrawerHeader,
+  DrawerOverlay,
+  Flex,
+  IconButton,
+  Text,
+} from "@fidesui/react";
+import { CloseSolidIcon } from "common/Icon";
+import { ExecutionLog } from "privacy-requests/types";
+import React, { useState } from "react";
+
+import EventError from "./EventError";
+import EventLog from "./EventLog";
+
+export type EventData = {
+  key: string;
+  logs: ExecutionLog[];
+};
+
+type EventDetailsProps = {
+  eventData: EventData | undefined;
+};
+
+const EventDetails = ({ eventData }: EventDetailsProps) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const [isViewingError, setViewingError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const openErrorPanel = (message: string) => {
+    setErrorMessage(message);
+    setViewingError(true);
+  };
+
+  const closeErrorPanel = () => {
+    setViewingError(false);
+  };
+
+  const closeDrawer = () => {
+    if (isViewingError) {
+      closeErrorPanel();
+    }
+
+    onClose();
+  };
+
+  const headerText = isViewingError ? "Event detail" : "Event log";
```

Comment:
Did we think about having this popover update the URL? I could imagine a user wanting to share event logs for request failures for instance, as that's where the trace will be.

Reply:
I don't think so. That's a good idea but it won't work right now because our current BE router isn't able to resolve dynamic urls. Once we decide on a solution for dynamic urls we will be able support features like this.

Are you okay with adding this in a future PR once that BE supports this?

Reply:
Sure! Have made a [ticket](https://github.com/ethyca/fidesops/issues/1027)